### PR TITLE
GISAID WDL workflow

### PR DESCRIPTION
**Pull request by kislyuk**

### CODE REVIEW

**Comment by ttung** [comment]
Is this still draft?

**Comment by kislyuk** [comment]
Still being integration tested but comments welcome

File: workflows/pangolin/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    gcc \
+    && locale-gen en_US.UTF-8
+
+RUN git clone --depth 1 git://github.com/chanzuckerberg/aspen
+RUN python3 -m venv /aspen/.venv
+RUN bash -c "source /aspen/.venv/bin/activate; pip3 install /aspen/src/backend /aspen/src/backend/third-party/sqlalchemy-enum-tables"
+
+RUN git clone --depth 1 git://github.com/nextstrain/ncov-ingest
+RUN python3 -m venv /ncov-ingest/.venv
+RUN bash -c "source /ncov-ingest/.venv/bin/activate; pip install -U pip pipenv; pipenv install"
```
Comment (phoenixAja) [inline workflows/pangolin/Dockerfile:41]: we don't need ncov ingest for pangolin, so i think this can be removed

Comment (phoenixAja) [reply workflows/pangolin/Dockerfile:41]: also need pangolin to be installed, [see this file for how i installed it in the run pangolin shell script](https://github.com/chanzuckerberg/aspen/blob/phoenix/pagolin-run/src/backend/aspen/workflows/pangolin/run_pangolin.sh)

should probably add installing pangolin to an entrypoint, as we want to make sure we have the most recent version when running

File: workflows/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
+FROM ubuntu:20.04
```
Comment (ttung) [inline workflows/Dockerfile:1]: so this model implies that the docker image needs to be rebuilt periodically?

Comment (ttung) [reply workflows/Dockerfile:1]: And why not build off the nextstrain image?  That ensures that the libraries required for nextstrain are always installed correctly.

File: workflows/gisaid.wdl [hunk 3]
```diff
@@ -0,0 +1,275 @@
+version 1.1
+
+workflow LoadGISAID {
+    input {
+        String docker_image_id = "aspen-gisaid"
+        String aws_region = "us-west-2"
+        String db_data_bucket = "aspen-db-data-dev"
```
Comment (ttung) [inline workflows/gisaid.wdl:7]: how does this work if the bucket changes depending on prod/staging?

File: workflows/pangolin/Dockerfile (hunk 1 above)
Comment (jgadling) [reply workflows/pangolin/Dockerfile:41]: Yeah, let's work together on getting a functional pangolin dockerfile. I think this is just a starting point.

File: workflows/gisaid.wdl [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        String gisaid_ndjson_export_url = "https://www.epicov.org/epi3/3p/exp3/export/export.json.bz2"
+        String gisaid_ndjson_staging_bucket = "aspen-data-dev"
+        String gisaid_ndjson_staging_key = "raw_gisaid_dump/test.zst"
```
Comment (ttung) [inline workflows/gisaid.wdl:10]: why is this test.xst?  also, this implies that every run will write to the same place.

File: workflows/Dockerfile (hunk 2 above)
Comment (kislyuk) [reply workflows/Dockerfile:1]: I usually prefer to build from scratch and use plain ubuntu and versioned PyPI dependencies where possible, but I'm not opposed to using an upstream Docker image instead as long as it's built reproducibly. Where can I get the upstream nextstrain dockerfile? I can give it a try to see if it works as expected.

File: workflows/gisaid.wdl (hunk 3 above)
Comment (kislyuk) [reply workflows/gisaid.wdl:7]: This is a test placeholder; it is expected to be overridden at runtime.

File: workflows/gisaid.wdl (hunk 4 above)
Comment (kislyuk) [reply workflows/gisaid.wdl:10]: Also a test placeholder, expected to be overridden at runtime.

Comment (kislyuk) [reply workflows/gisaid.wdl:10]: In general I agree these placeholder defaults should be removed, and test values should be set in the test suite instead.

However in the particular case of gisaid dump caching (see the RefreshGISAID task - the idea is not to download and decompress the large upstream file every time if it has not changed), if this input changes, that task will need a separate pointer to the most recently downloaded copy of the file as well, in order for the caching logic to work.

**Pull request merged**